### PR TITLE
Dennis nc news 5

### PR DESCRIPTION
**Pull request by DennisTockan**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -34,13 +34,36 @@ describe("GET /api/topics", () => {
...
+          expect(article).toHaveProperty("title", expect.any(String));
+          expect(article).toHaveProperty("article_id", expect.any(Number));
+          expect(article).toHaveProperty("topic", expect.any(String));
+          expect(article).toHaveProperty("created_at", expect.any(String));
+          expect(article).toHaveProperty("votes", expect.any(Number));
+          expect(article).toHaveProperty("article_img_url", expect.any(String));
+          expect(article).toHaveProperty("comment_count", expect.any(Number));
+        });
+      });
+  });
```
Comment (NRMSMITH) [inline __tests__/app.test.js:71]: This ticket also asks for it to be sorted by created_at desc :) Have a look into jest-sorted to help you with that!

File: controllers/articles.controllers.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
+const { selectAllArticles } = require("../models/articles.model");
+
+exports.getAllArticles = (_, res, next) => {
+  selectAllArticles().then((articles) => {
+    // console.log(article, 'article')
```
Comment (NRMSMITH) [inline controllers/articles.controllers.js]: make sure to delete console.logs

File: models/articles.model.js [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+    .then(({ rows }) => {
+      if (!rows.length) {
+        return Promise.reject({
+          status: 404,
+          message: "Not Found",
+        });
+      }
+
+      return rows.map((row) => {
+        row.comment_count = Number(row.comment_count);
```
Comment (NRMSMITH) [inline models/articles.model.js]: I don't think this map is necessary - have a look at what rows looks like upon its return

**Review by NRMSMITH — CHANGES_REQUESTED** [review]
A few points to focus on

File: __tests__/app.test.js [hunk 4]
```diff
@@ -34,13 +34,38 @@ describe("GET /api/topics", () => {
...
-      expect(body).toEqual(apiEndpoints)
-    })
-  })
-})
\ No newline at end of file
+      .get("/api/articles")
+      .expect(200)
+      .then(({ body }) => {
+        const { articles } = body;
+        console.log(articles)
```
Comment (hevdevs) [inline __tests__/app.test.js:56]: remove console.log before merge

File: __tests__/app.test.js [hunk 5]
```diff
@@ -34,13 +34,38 @@ describe("GET /api/topics", () => {
...
+        expect(articles).toHaveLength(5);
```
Comment (hevdevs) [inline __tests__/app.test.js:57]: is this the correct number of articles?

**Review by hevdevs — APPROVED** [review]
Hey Dennis! Code is looking great.  🎉
Please can you add info on the get /api/articles to your endpoints.json if you haven't already? Also please remove the highlighted console.log before merging and check the test data aligns with the test assertion that I've highlighted. No need to put in another PR once done

**Pull request merged**